sql/semisync_master.h

```diff
@@ -697,6 +707,12 @@ class Repl_semi_sync_master
   /*called before reset master*/
   int before_reset_master();
 
+  /**
+    With lock() held, call Active_tranx::clear_active_tranx_nodes()
+    with Active_tranx::find_acked_tranx_node() if there is one
+  */
```

Comment:
For header-level comments, it's better to summarize the purpose of a function, rather than how it is implemented. Implementations often change, and these types of comments are not usually changed with them, easily leading to inconsistencies and harder to maintain code. The purpose of a function shouldn't ever really change though.